go/ct/spc/specification_test.go

```diff
@@ -373,3 +377,83 @@ func TestSumWithOverflow(t *testing.T) {
 		})
 	}
 }
+
+func TestSpecification_OpsWithDynamicCostHandleOverflowSizes(t *testing.T) {
```

Comment:
I like the general structure of this test, but I am not too confident about the code filtering out rules to be tested. I do not think there is a need for this. We can just test all rules, as it is done here:
```

func TestSpecification_OpsWithDynamicCostHandleOverflowSizes(t *testing.T) {

	indexOfMemParam := map[vm.OpCode][]int{
		vm.MCOPY:          {0, 1, 2},
		vm.CODECOPY:       {0, 2},
		vm.CALLDATACOPY:   {0, 2},
		vm.RETURNDATACOPY: {0, 1, 2},
		vm.REVERT:         {0, 1},
		vm.CREATE:         {1, 2},
		vm.CREATE2:        {1, 2},
		vm.EXTCODECOPY:    {1, 3},
		vm.CALLCODE:       {3, 4, 5, 6},
		vm.SHA3:           {0, 1},
		vm.STATICCALL:     {2, 3, 4, 5},
	}
	testValues := []uint64{st.MaxMemoryExpansionSize, st.MaxMemoryExpansionSize + 1, math.MaxUint64}

	rnd := rand.New(0)

	allRules := getAllRules()
	for op := range indexOfMemParam {
		for revision := common.MinRevision; revision <= common.NewestSupportedRevision; revision++ {
			for _, overflowPosition := range indexOfMemParam[op] {
				for _, value := range testValues {
					t.Run(fmt.Sprintf("%v_%v_%v_%v", op, revision, overflowPosition, value), func(t *testing.T) {

						condition := rlz.And(
							rlz.IsRevision(revision),
							rlz.Eq(rlz.Status(), st.Running),
							rlz.Eq(rlz.Op(rlz.Pc()), vm.OpCode(op)),
							rlz.Eq(rlz.Param(overflowPosition), common.NewU256(value)),
						)

						for _, i := range indexOfMemParam[op] {
							if i != overflowPosition {
								condition = rlz.And(condition, rlz.Eq(rlz.Param(i), common.NewU256(1)))
							}
						}

						generator := gen.NewStateGenerator()
						condition.Restrict(generator)

						state, err := generator.Generate(rnd)
						if err != nil {
							t.Fatalf("failed to generate a random state: %v", err)
						}

						for _, rule := range allRules {
							match, err := rule.Condition.Check(state)
							if err != nil {
								t.Fatalf("failed to check rule condition for %v: %v", rule.Name, err)
							}
							if match {
								result := state.Clone()
								rule.Effect.Apply(result)
								if want, got := st.Failed, result.Status; want != got {
									t.Fatalf("Rule %s does not fail with overflow parameters, got %v", rule.Name, got)
								}
								result.Release()
							}
						}
					})
				}
			}
		}
	}
}
```
Wdyt?

Reply:
I didn't take this approach originally because I wanted to reduce the number of `rule.Condition.Check`, but I agree it looks cleaner this way.